synapse/api/auth.py

```diff
@@ -629,7 +629,13 @@ def get_user_by_access_token(self, token, rights="access"):
         except AuthError:
             # TODO(daniel): Remove this fallback when all existing access tokens
             # have been re-issued as macaroons.
+            if self.hs.config.expire_access_token:
+                raise AuthError(
+                    401,
+                    "access token has expired"
+                )
```

Comment:
Yes, I agree you cannot really tell which Caveat(restriction) is failing here. But I didn't understand your last comment very well. Could you please explain a bit. Do you mean you would just raise here with no error message?